File: 이지수/Swift/Programmers_67256.swift
import Foundation

func solution(_ numbers:[Int], _ hand:String) -> String {
    let priority = hand == "left" ? 0 : 1
    var position = [10, 12]
    var answer = ""
    numbers.map() {$0 != 0 ? $0 : 11}.forEach() { num in
        switch num%3 {
        case 1 :
            position[0] = num
            answer += "L"
        case 0 :
            position[1] = num
            answer += "R"
        default :
            let pos = closer(num, position[0], position[1], priority)
            position[pos] = num
            answer += pos == 0 ? "L" : "R"
        }
    }
    return answer
}

func closer(_ num: Int, _ a: Int, _ b: Int, _ d: Int) -> Int {
    let distance = (abs((num-1)/3-(a-1)/3)+abs((num-1)%3-(a-1)%3), abs((num-1)/3-(b-1)/3)+abs((num-1)%3-(b-1)%3))
    if distance.0 > distance.1 { return 1 }
    else if distance.0 < distance.1 { return 0 }
    else { return d }
}

File: 이지수/Python/Programmers_67256.py
def solution(numbers, hand):
    priority = 0 if hand == "left" else 1
    numbers = [num if num != 0 else 11 for num in numbers]
    p = [10, 12]
    answer = ''
    for num in numbers :
        n = num % 3 
        if n == 1 : p[0] = num; answer+='L'
        elif n == 0 : p[1] = num; answer+='R'
        else : 
            pos = closer(diff(num, p[0]), diff(num, p[1]), priority)
            p[pos] = num; answer+='L' if pos == 0 else 'R'
    return answer

def diff(num, p) :
    num -= 1; p -= 1
    return abs(num//3-p//3) + abs(p%3-num%3)

def closer(a, b, d) :
    if a != b : return 0 if a < b else 1
    else : return d
